clang/test/SemaCXX/warn-unused-variables.cpp

```diff
@@ -1,5 +1,5 @@
 // RUN: %clang_cc1 -fsyntax-only -Wunused-variable -Wunused-label -Wno-c++1y-extensions -verify %s
-// RUN: %clang_cc1 -fsyntax-only -Wunused-variable -Wunused-label -Wno-c++1y-extensions -verify -std=gnu++11 %s
+// RUN: %clang_cc1 -fsyntax-only -Wunused-variable -Wunused-label -Wno-c++1y-extensions -verify -std=gnu++17 %s
```

Comment:
I think that would give an error when the gnu++11 check is run?

Reply:
> I think that would give an error when the gnu++11 check is run?

It seems to fail on all standards below C++14.

Reply:
That seems fair. How do you think I should proceed? Should I write a new test file?

Reply:
I agree with @Endilll.
We should understand _why_ the behavior in older language modes is different.
I would recommend trying to use a debugger to try to see why we are not passing in that code path.
The difference in behavior across language modes looks like a bug that should be fixed, rather than a change in expected behavior

(in all language modes, if a constructor is called, that constructor might have side effects)

Reply:
Got it! I am on it, please give me some time and I'll report back.

Reply:
OK, the difference in behaviour seems to stem from the fact that in C++-11 mode and C++-14 mode, the constructors are marked as not elidable(clang/lib/Sema/SemaDecl.cpp, line 2049). This causes it to return false when these constructors which are created by copy initialization are encountered, specifically while passing in C++ standard modes lower than C++-17